modyn/tests/evaluator/internal/test_pytorch_evaluator.py

```diff
@@ -184,108 +171,93 @@ def test_load_model():
-        query_status_queue, response_queue, mp.Queue(), "trained_model.modyn", True
+        "trained_model.modyn", True, metric_queue, not_failed_interval_ids, metric_jsons
     )
+    evaluator.evaluate()
 
-    query_status_queue.put("INVALID MESSAGE")
-    timeout = 5
-    elapsed = 0
-    while query_status_queue.empty():
-        sleep(0.1)
-        elapsed += 0.1
-
-        if elapsed >= timeout:
-            raise TimeoutError("Did not reach desired queue state within time limit.")
-
-    with pytest.raises(ValueError):
-        evaluator.evaluate()
-
-    elapsed = 0
-    while not (query_status_queue.empty() and response_queue.empty()):
-        sleep(0.1)
-        elapsed += 0.1
-
-        if elapsed >= timeout:
-            raise TimeoutError("Did not reach desired queue state within time limit.")
-
-
-@patch("modyn.evaluator.internal.pytorch_evaluator.EvaluationDataset", MockEvaluationDataset)
-@patch.object(PytorchEvaluator, "_load_state")
-def test_evaluate(load_state_mock: MagicMock):
-    query_status_queue = mp.Queue()
-    response_queue = mp.Queue()
-    metric_result_queue = mp.Queue()
-    evaluator: PytorchEvaluator = get_mock_evaluator(
-        query_status_queue, response_queue, metric_result_queue, "trained_model.modyn", True
+    prepare_dataloader_mock.assert_has_calls(
+        [
+            call(ANY, None, None),
+            call(ANY, None, 100),
+            call(ANY, 100, 200),
+            call(ANY, 200, None),
+        ]
     )
 
-    query_status_queue.put(EvaluatorMessages.STATUS_QUERY_MESSAGE)
-    timeout = 2
-    elapsed = 0
-
-    while query_status_queue.empty():
-        sleep(0.1)
-        elapsed += 0.1
-
-        if elapsed >= timeout:
-            raise TimeoutError("Did not reach desired queue state within timelimit.")
-
-    evaluator.evaluate()
-    assert evaluator._num_samples == 100
-    elapsed = 0
-    while not query_status_queue.empty():
-        sleep(0.1)
-        elapsed += 0.1
-
-        if elapsed >= timeout:
-            raise TimeoutError("Did not reach desired queue state within timelimit.")
-
-    elapsed = 0
-    while True:
-        if not platform.system() == "Darwin":
-            if response_queue.qsize() == 1 and metric_result_queue.qsize() == 1:
-                break
-        else:
-            if not response_queue.empty() and not metric_result_queue.empty():
-                break
-
-        sleep(0.1)
-        elapsed += 0.1
-
-        if elapsed >= timeout:
-            raise AssertionError("Did not reach desired queue state after 5 seconds.")
-
-    status = response_queue.get()
-    assert status["num_batches"] == 0
-    assert status["num_samples"] == 0
-
-    # accuracy
-    metric_name, metric_result = metric_result_queue.get()
-    assert metric_name == Accuracy(AccuracyMetricConfig()).get_name()
-    assert metric_result == pytest.approx(1)
+    expected_accuracies = [1.0, 0.01, 0.0, 0.5]
+    expected_f1scores = [0.5, ANY, 0.0, 1 / 3]
```

Comment:
the second f1 score is a number not so easily computed from outside. I just trust the code